google/cloud/pubsub/internal/publisher_round_robin_test.cc

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsub/internal/publisher_round_robin.h"
+#include "google/cloud/pubsub/testing/mock_publisher_stub.h"
+#include "google/cloud/testing_util/assert_ok.h"
+#include "absl/memory/memory.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+inline namespace GOOGLE_CLOUD_CPP_PUBSUB_NS {
+namespace {
+
+using ::testing::Return;
+
+TEST(PublisherRoundRobinTest, CreateTopic) {
```

Comment:
Opt nit: These indeed check that each mock is used once, but "round robin" is a slightly stronger condition.

Reply:
Fair enough.  I am not sure if the new tests is what you had in mind, if not, let me what are you thinking about.

Reply:
Yes, something like `InSequence` and a multiplier on the call loop was what I was thinking.  Thanks.